### PR TITLE
Added missing unit tests for deleting items from list controls

### PR DESCRIPTION
**Pull request by SergeySmirnov-Akvelon**
## Proposed changes
- Added missing unit tests for deleting items from list controls

## Customer Impact
- No 

## Regression? 

- No

## Risk
- Minimal

## Test environment(s) 
- Microsoft Windows [Version 10.0.19044.1466]
- .NET Core SDK: 7.0.0-preview.2.22113.2

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6716)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 1]
```diff
@@ -6112,6 +6112,176 @@ public void ListBox_ToString_InvokeWithItems_ReturnsExpected(string item1, strin
...
+            Assert.Equal(1, listBox.SelectedItems.Count);
+            Assert.Equal(createControl, listBox.IsHandleCreated);
+        }
+
+        [WinFormsTheory]
+        [InlineData(SelectionMode.MultiSimple, true)]
+        [InlineData(SelectionMode.MultiSimple, false)]
+        [InlineData(SelectionMode.MultiExtended, true)]
+        [InlineData(SelectionMode.MultiExtended, false)]
+        public void ListBox_Remove_NotSelectedItems_IfseveralItemSelected(SelectionMode mode, bool createControl)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6161]: `several` should be capitalized

**Review by Tanya-Solyanik — COMMENTED** [review]
I added some questions

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs [hunk 2]
```diff
@@ -396,6 +396,164 @@ public void CheckedListBox_OnDrawItem_LargeEIndexNotEmpty_Success()
...
+
+            for (int i = 0; i < 2; i++)
+            {
+                Assert.Equal(2, checkedListBox.CheckedIndices.Count);
+                Assert.Equal(2, checkedListBox.CheckedItems.Count);
+
+                checkedListBox.Items.Remove(checkedListBox.Items[3]);
+                checkedListBox.Items.Remove(checkedListBox.Items[2]);
+
+                count -= 2;
```
[the remark this answers is not in the document]
Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/tests/UnitTests/CheckedListBoxTests.cs:509]: I think you are right. I just wanted to check that when working with several selected elements at once, everything also works correctly.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 3]
```diff
@@ -6112,6 +6112,158 @@ public void ListBox_ToString_InvokeWithItems_ReturnsExpected(string item1, strin
...
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6159]: public void ListBox_Remove_NotSelectedItems_IfSeveralItemSelected(SelectionMode mode, bool createControl)

Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6159]:
```suggestion
        public void ListBox_Remove_NotSelectedItems_IfSeveralItemSelected(SelectionMode mode, bool createControl)
```

**Review by RussKie — DISMISSED** [review]

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs (hunk 3 above)
Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs]: Thank you. Fixed

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good, thank you!

**Pull request merged**